lib/mbaasRequest/mbaasRequest.js

```diff
@@ -177,7 +177,7 @@ function responseHandler(params, cb) {
     if (err || (httpResponse && httpResponse.statusCode >= 400)) {
 
       //There is a specific error message if an environment is unreachable
-      if (httpResponse && httpResponse.statusCode) {
+      if (httpResponse && (httpResponse.statusCode || err.code)) {
```

Comment:
There is a `if (httpResponse.statusCode === 503) {` inside this if statement below, which could cause a problem. I think these if statements will need to be broken up.